ui/app/models/node.js

```diff
@@ -153,4 +153,17 @@ export default class Node extends Model {
   cancelDrain() {
     return this.store.adapterFor('node').cancelDrain(this);
   }
+
+  async addMeta(newMeta) {
+    let metaResponse = await this.store
+      .adapterFor('node')
+      .addMeta(this, newMeta);
+
+    if (!this.meta) {
+      this.set('meta', this.store.createFragment('structured-attributes'));
+    }
```

Comment:
`null !== []` so we create a fragment with raw/recompute/structured methods on it if there's nothing at all.

Reply:
`question`: So this is a defensive technique we're using before calling `recomputeRawProperties` which then computes the derived state for `structured`?

Where are we comparing `null` to `[]`? Is this something we're doing for the template layer?

Reply:
If we have no metadata, we get a null back. If we had metadata but use dynamic node metadata to overwrite it all to null values, we get `[]` back. In both cases we need to have actual `structured-attributes`.